cpp/test/sg/rf_depth_test.cu

```diff
@@ -0,0 +1,177 @@
+class RfClassifierDepthTest : public ::testing::TestWithParam<int> {
+ protected:
+  void basicTest() {
+    const int max_depth = ::testing::TestWithParam<int>::GetParam();
+    params = RfInputs<T>{5000, 10, 1, 1.0f, 1.0f, max_depth, -1, false,
+                         false, 8, SPLIT_ALGO::GLOBAL_QUANTILE, 2, 0.0, 2,
+                         CRITERION::ENTROPY};
+
+    DecisionTree::DecisionTreeParams tree_params;
+    set_tree_params(tree_params, params.max_depth, params.max_leaves,
+                    params.max_features, params.n_bins, params.split_algo,
+                    params.min_rows_per_node, params.min_impurity_decrease,
+                    params.bootstrap_features, params.split_criterion, false);
+    RF_params rf_params;
+    set_all_rf_params(rf_params, params.n_trees, params.bootstrap,
+                      params.rows_sample, -1, params.n_streams, tree_params);
+
+    int data_len = params.n_rows * params.n_cols;
+    allocate(data, data_len);
+    allocate(labels, params.n_rows);
+
+    cudaStream_t stream;
+    CUDA_CHECK(cudaStreamCreate(&stream));
+
+    // Populate data (assume Col major)
+    std::mt19937 gen(0);
+    std::vector<T> data_h(data_len);
+    std::normal_distribution<> d{0, 1};
+    for (int col = 0; col < params.n_cols; ++col) {
+      for (int row = 0; row < params.n_rows; ++row) {
+        data_h[row + col * params.n_rows] = d(gen);
+      }
+    }
+    updateDevice(data, data_h.data(), data_len, stream);
+
+    // Populate labels
+    labels_h.resize(params.n_rows);
+    for (int row = 0; row < params.n_rows; ++row) {
+      labels_h[row] = (data_h[row + 2 * params.n_rows]
+                         * data_h[row + 3 * params.n_rows] > 0.5);
+    }
+    preprocess_labels(params.n_rows, labels_h, labels_map);
+    updateDevice(labels, labels_h.data(), params.n_rows, stream);
+
+    forest = new typename ML::RandomForestMetaData<T, int>;
+    null_trees_ptr(forest);
+
+    cumlHandle handle(rf_params.n_streams);
+    handle.setStream(stream);
+
+    fit(handle, forest, data, params.n_rows, params.n_cols, labels,
+        labels_map.size(), rf_params);
+
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+  }
+
+  void SetUp() override { basicTest(); }
+
+  void TearDown() override {
+    labels_h.clear();
+    labels_map.clear();
+
+    CUDA_CHECK(cudaFree(labels));
+    CUDA_CHECK(cudaFree(data));
+    delete forest;
+  }
+
+ protected:
+  RfInputs<T> params;
+  T *data;
+  int* labels;
+  std::vector<int> labels_h;
+  std::map<int, int> labels_map;
+    // unique map of labels to int vals starting from 0
+
+  RandomForestMetaData<T, int>* forest;
+};
+
+template <typename L, typename T>
+int MaxDepthOfDecisionTree(const DecisionTree::TreeMetaDataNode<T, L>* tree) {
+  const auto& node_array = tree->sparsetree;
+  std::queue<std::pair<int, int>> q;  // (node ID, depth)
+  // Traverse the tree breadth-first
+  int initial_depth = 1;
+    // Currently, cuML starts counting depth by 1.
```

Comment:
Note: we'd like to change `initial_depth` to 0, to be consistent with scikit-learn RF.